### ISSUE TITLE
Error bars on barplots from mixed models

### ISSUE THREAD

**Issue by Wortly** [kassambara/ggpubr#403]
Wondering if there is a way to add custom error bars?  The SEM from mixed models analysis is not necessarily based on arithmetic means.  This creates an issue when you are using ggpubr and the errorbars are based on arithmetic means.  Is there a workaround?

**Comment by kassambara** [comment]
Yes — precompute the summary (estimated means + your model's SEMs, e.g. from `emmeans::emmeans(model)`), plot the means with `ggbarplot()` (no `add=`), and draw your own `geom_errorbar()` from the model columns:

``` r
suppressWarnings(suppressPackageStartupMessages(library(ggpubr)))
# Model-based means + SEMs (e.g. from emmeans/lme4): precompute, then draw your own bars.
summ <- data.frame(dose = factor(c(0.5, 1, 2)),
                   emmean = c(10.6, 19.7, 26.1), SE = c(1.2, 0.8, 1.5))
ggbarplot(summ, x = "dose", y = "emmean", fill = "dose", palette = "npg") +
  geom_errorbar(aes(ymin = emmean - SE, ymax = emmean + SE), width = 0.2)
```

![](https://i.imgur.com/Jxb3zGO.png)

Avoid `add = "mean_se"` here — it recomputes an *arithmetic* SE from the raw data rather than using your model-based SEM.

Closing as resolved — reopen if anything's unclear. 🙌
